src/main/java/com/nike/cerberus/endpoints/authentication/AuthenticateStsIdentity.java

```diff
@@ -0,0 +1,118 @@
+import com.nike.cerberus.domain.AuthTokenResponse;
+import com.nike.cerberus.error.DefaultApiError;
+import com.nike.cerberus.service.AuthenticationService;
+import com.nike.cerberus.service.EventProcessorService;
+import com.nike.riposte.server.http.RequestInfo;
+import com.nike.riposte.server.http.ResponseInfo;
+import com.nike.riposte.server.http.StandardEndpoint;
+import com.nike.riposte.util.AsyncNettyHelper;
+import com.nike.riposte.util.Matcher;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpMethod;
+import okhttp3.MediaType;
+
+import javax.inject.Inject;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+
+import static com.nike.cerberus.aws.sts.AwsStsHttpHeaders.*;
+import static com.nike.cerberus.endpoints.AuditableEventEndpoint.auditableEvent;
+
+/**
+ * Authentication endpoint for IAM roles.  If valid, a client token that is encrypted via KMS is returned.  The
+ * IAM role will be the only role capable of decrypting the client token via KMS.
+ */
+public class AuthenticateStsIdentity extends StandardEndpoint<Void, AuthTokenResponse> {
+
+    private final AuthenticationService authenticationService;
+    private final EventProcessorService eventProcessorService;
+    private final AwsStsClient awsStsClient;
+    private static final MediaType DEFAULT_MEDIA_TYPE = MediaType.parse("application/json");
+
+    @Inject
+    public AuthenticateStsIdentity(AuthenticationService authenticationService,
+                                   EventProcessorService eventProcessorService,
+                                   AwsStsClient awsStsClient) {
+
+        this.authenticationService = authenticationService;
+        this.eventProcessorService = eventProcessorService;
+        this.awsStsClient = awsStsClient;
+    }
+
+    @Override
+    public CompletableFuture<ResponseInfo<AuthTokenResponse>> execute(final RequestInfo<Void> request,
+                                                                        final Executor longRunningTaskExecutor,
+                                                                        final ChannelHandlerContext ctx) {
+        return CompletableFuture.supplyAsync(
+                AsyncNettyHelper.supplierWithTracingAndMdc(() -> authenticate(request), ctx),
+                longRunningTaskExecutor
+        );
+    }
+
+    private ResponseInfo<AuthTokenResponse> authenticate(RequestInfo<Void> request) {
+        final String date = getDate(request);
+        final String headerXAmzDate = getHeaderXAmzDate(request);
+        final String headerXAmzSecurityToken = getHeaderXAmzSecurityToken(request);
+        final String headerAuthorization = getHeaderAuthorization(request);
+
+        if (date == null || headerAuthorization == null || headerXAmzDate == null || headerXAmzSecurityToken == null) {
+            throw new ApiException(DefaultApiError.GENERIC_BAD_REQUEST);
```

Comment:
This error message / error code is not useful. Shouldn't you create a new error?